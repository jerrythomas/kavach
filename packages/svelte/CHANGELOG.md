# @kavach/svelte

## 1.0.0-next.6

### Patch Changes

- Added store for auth status and used alerts to display status on demo page.
- Updated dependencies
  - @kavach/cookie@1.0.0-next.6
  - @kavach/core@1.0.0-next.6
  - @kavach/deflector@1.0.0-next.6
  - @kavach/logger@1.0.0-next.6

## 1.0.0-next.5

### Patch Changes

- chore: release 1.0.0-next.4
- Split core package into logger and deflector
- 1a35ad0: Resolved linter errors.
- Releasing the next version.
- Updated dependencies
- Updated dependencies
- Updated dependencies [1a35ad0]
- Updated dependencies
  - @kavach/cookie@1.0.0-next.5
  - @kavach/core@1.0.0-next.5
  - @kavach/deflector@1.0.0-next.5
  - @kavach/logger@1.0.0-next.5
