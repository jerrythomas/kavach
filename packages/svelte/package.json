{
	"name": "@kavach/svelte",
	"version": "1.0.0-next.2",
	"description": "Drop in security for web apps.",
	"author": "Jerry Thomas <me@jerrythomas.name>",
	"license": "MIT",
	"main": "index.js",
	"svelte": "src/index.js",
	"module": "src/index.js",
	"types": "dist/index.d.ts",
	"type": "module",
	"repository": {
		"type": "git",
		"url": "https://github.com/jerrythomas/kavach",
		"directory": "packages/kavach"
	},
	"prettier": "@jerrythomas/prettier-config",
	"publishConfig": {
		"access": "public"
	},
	"scripts": {
		"prepublishOnly": "tsc",
		"lint": "prettier --check --plugin-search-dir=. . && eslint .",
		"format": "prettier --write --plugin-search-dir=. .",
		"test:ci": "vitest run",
		"test:ui": "vitest --ui",
		"test": "vitest",
		"coverage": "vitest run --coverage",
		"upgrade": "pnpm upgrade"
	},
	"devDependencies": {
		"@jerrythomas/prettier-config": "^1.0.0",
		"@sveltejs/vite-plugin-svelte": "^2.0.2",
		"@testing-library/svelte": "^3.2.2",
		"@vitest/ui": "~0.12.10",
		"c8": "^7.12.0",
		"eslint-config-svelte": "workspace:*",
		"js-yaml": "^4.1.0",
		"jsdom": "^19.0.0",
		"shared-config": "workspace:^1.0.0",
		"svelte": "^3.55.0",
		"typescript": "^4.9.4",
		"vite": "^4.0.3",
		"vitest": "~0.19.1"
	},
	"exports": {
		"./package.json": "./package.json",
		".": {
			"types": "./types/index.d.ts",
			"import": "./src/index.js"
		}
	},
	"dependencies": {
		"@kavach/cookie": "workspace:1.0.0-next.2",
		"@kavach/core": "workspace:*",
		"@svelte-spice/form": "1.0.0-next.3",
		"@svelte-spice/input": "1.0.0-next.3",
		"ramda": "^0.28.0"
	}
}
