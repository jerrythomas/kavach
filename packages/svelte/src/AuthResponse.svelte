<script>
	export let error = false
	export let message = ''
</script>

{#if message.length > 0}
	<alert class="flex flex-col w-full" class:hasError={error}>
		{message}
	</alert>
{/if}
