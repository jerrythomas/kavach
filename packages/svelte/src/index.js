export * from './kavach'
export * from './request'
export * from './endpoints'

export { default as AuthButton } from './AuthButton.svelte'
export { default as AuthProvider } from './AuthProvider.svelte'
export { default as AuthGroup } from './AuthGroup.svelte'
export { default as AuthError } from './AuthError.svelte'
export { default as AuthResponse } from './AuthResponse.svelte'
export { default as AuthHandler } from './AuthHandler.svelte'
export { default as Logout } from './Logout.svelte'
