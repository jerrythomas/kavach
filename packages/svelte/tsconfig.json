{
  "extends": "shared-config/jsconfig.json",
  "include": ["src/**/*.js", "src/**/*.svelte"],
  "compilerOptions": {
    "outDir": "dist"
  }
}
