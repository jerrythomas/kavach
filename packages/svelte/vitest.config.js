import { defineConfig } from 'vite'
import { svelte } from '@sveltejs/vite-plugin-svelte'
import { testConfig } from 'shared-config'

testConfig.coverage.provider = 'v8'
export default defineConfig({
	plugins: [svelte({ hot: !process.env.VITEST })],
	test: testConfig
})
