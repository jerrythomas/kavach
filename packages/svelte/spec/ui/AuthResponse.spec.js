import { describe, expect, it, beforeEach } from 'vitest'
import { cleanup, render } from '@testing-library/svelte'
import { tick } from 'svelte'
import AuthResponse from '../../src/ui/AuthResponse.svelte'

describe('AuthResponse.svelte', () => {
	beforeEach(() => {
		cleanup()
	})

	it('should render response with changes', async () => {
		const { container, component } = render(AuthResponse)
		expect(container).toMatchSnapshot()
		component.$$set({ message: 'Successfully logged in' })
		await tick()
		expect(container).toMatchSnapshot()
	})

	it('should render response with error', () => {
		const { container } = render(AuthResponse, {
			props: { message: 'An error occurred', error: true }
		})
		expect(container).toMatchSnapshot()
	})
})
