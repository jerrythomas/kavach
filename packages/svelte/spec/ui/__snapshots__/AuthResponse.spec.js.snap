// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AuthResponse.svelte > should render response with changes 1`] = `
<body>
  <div>
    
  </div>
</body>
`;

exports[`AuthResponse.svelte > should render response with changes 2`] = `
<body>
  <div>
    <alert
      class="flex flex-col w-full"
    >
      Successfully logged in
    </alert>
    
  </div>
</body>
`;

exports[`AuthResponse.svelte > should render response with error 1`] = `
<body>
  <div>
    <alert
      class="flex flex-col w-full hasError"
    >
      An error occurred
    </alert>
    
  </div>
</body>
`;
